sys/vane/clay.hoon

```diff
@@ -2109,1645 +2647,1089 @@
-            =<  t
-            ~|  [%letdom let=let.dom hit=hit.dom hut=(~(run by hut.ran) ,~)]
-            ~|  [%getdom (~(get by hit.dom) let.dom)]
-            %-  aeon-to-yaki
-            let.dom
-          [~ let.dom]
-        $(let.dom (dec let.dom))
+        |=  $:  nex/(unit aeon)
+                old/(map (pair care path) cach)
+                new/(map (pair care path) cach)
+            ==
+        ^+  ..start-request
+        %+  duce  for
+        ^-  rove
+        ?:  ?=($mult -.rav)
+          [-.rav p.rav nex old new]
+        :^  -.rav  p.rav  nex
+        =+  ole=~(tap by old)
+        ?>  (lte (lent ole) 1)
+        ?~  ole  ~
+        q:(snag 0 `(list (pair (pair care path) cach))`ole)
       ::
-          $tas  (~(get by lab.dom) p.lok)
-          $ud   ?:((gth p.lok let.dom) ~ [~ p.lok])
-      ==
-    ::
-    ::  Convert a map of paths to data into an ankh.
+      ++  respond                                       ::  send changes
+        |=  res/(map mood (each cage lobe))
+        ^+  ..start-request
+        ?:  ?=($mult -.rav)  (blas hen ~(key by res))
+        ?>  ?=({* ~ ~} res)
+        (blab hen n.res)
+      ::
+      ++  know  |=({(pair care path) c/cach} ?=(^ c))   ::  know about file
+      ::
+      ++  read-all-at                                   ::  files at case, maybe
+        |=  cas/case
+        %-  ~(gas by *(map (pair care path) cach))
+        =/  req/(set (pair care path))
+          ?:  ?=($mult -.rav)  q.p.rav
+          [[p.p.rav r.p.rav] ~ ~]
+        %+  turn  ~(tap by req)
+        |=  {c/care p/path}
+        ^-  (pair (pair care path) cach)
+        [[c p] (aver for c cas p)]
+      --
     ::
-    ++  map-to-ankh
-      |=  hat/(map path (pair lobe cage))
-      ^-  ankh
-      ::  %-  cosh
-      %+  roll  ~(tap by hat)
-      |=  {{pat/path lob/lobe zar/cage} ank/ankh}
-      ^-  ankh
-      ::  %-  cosh
-      ?~  pat
-        ank(fil [~ lob zar])
-      =+  nak=(~(get by dir.ank) i.pat)
-      %=  ank
-        dir  %+  ~(put by dir.ank)  i.pat
-             $(pat t.pat, ank (fall nak *ankh))
+        $many
+      =+  nab=(case-to-aeon p.q.rav)
+      ?~  nab
+        ?>  =(~ (case-to-aeon q.q.rav))
+        (duce for [- p q ~]:rav)
+      =+  huy=(case-to-aeon q.q.rav)
+      ?:  &(?=(^ huy) |((lth u.huy u.nab) &(=(0 u.huy) =(0 u.nab))))
+        (blub hen)
+      =+  top=?~(huy let.dom u.huy)
+      =+  ear=(lobes-at-path:ze for top r.q.rav)
+      =.  +>.$
+        (bleb hen u.nab ?:(p.rav ~ `[u.nab top]))
+      ?^  huy
+        (blub hen)
+      =+  ^=  ptr  ^-  case
+          [%ud +(let.dom)]
+      (duce for `rove`[%many p.rav [ptr q.q.rav r.q.rav] ear])
+    ==
+  ::
+  ::  Continue committing
+  ::
+  ++  take-commit
+    |=  =sign
+    ^+  +>
+    ?~  act
+      ~|(%no-active-write !!)
+    ?.  ?=(%commit -.cad.u.act)
+      ~|(%active-not-write !!)
+    =/  c-res  (com.cad.u.act now ran sign)
+    =.  +>.$
+      =<  ?>(?=([~ * * * %commit *] act) .)  ::  TMI
```

Comment:
Could you instead add a `^-  note` to the gate that gets applied to each element?

Reply:
uh, or `^-  move`, maybe?

Reply:
oh, nevermind, the problem is with `:act` getting specialized to non-null. please ignore previous comments